core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -397,6 +399,28 @@ class SpecificUnitAutomation{
         else UnitAutomation().explore(unit, unit.getDistanceToTiles())
     }
 
+    fun automateGeneral(unit: MapUnit){
+        //try to follow nearby units. Do not garrison in city if possible
+        val militantToCompany = unit.civInfo.getCivUnits()
+                .firstOrNull { val tile = it.currentTile
+                    it.type.isLandUnit() && it.getMaxMovement() <= 2.0f && tile.civilianUnit==null
```

Comment:
The tile.civilianUnit==null means that if the military unit doesn't move, then the great general will try to move out next turn, maybe (tile.civilianUnit== null || tile.civilianUnit==unit)

Reply:
Agree.